tests/Regressions.hs

```diff
@@ -71,6 +76,48 @@ propEqAfterDelete (Keys keys) =
 mapFromKeys :: [Int] -> HM.HashMap Int ()
 mapFromKeys keys = HM.fromList (zip keys (repeat ()))
 
+------------------------------------------------------------------------
+-- Issue #254
```

Comment:
I've been able to reproduce the issue on my local machine by running these tests without your fix. Amazing! :)

These tests use (Simon Marlow's ?) weak-reference-based space leak detection technique, right? I've never actually seen it used in practice!

Could you possibly add a reference to an explanation of the technique?

Reply:
Done.